static/app/components/organizations/pageFilters/container.tsx

```diff
@@ -34,6 +34,12 @@ type Props = InitializeUrlStateProps & {
    * Custom alert message for the desynced filter state.
    */
   desyncedAlertMessage?: string;
+  /**
+   * When true, changes to page filters' value won't be saved to local storage, and will
+   * be forgotten when the user navigates to a different page. This is useful for local
+   * filtering contexts like in Dashboard Details.
+   */
+  disablePersistence?: boolean;
```

Comment:
We probably want a `useEffect` on this value so if it changes we update it right?

Reply:
Hmm as it's implemented here persistence gets updated every time `PageFiltersContainer` gets mounted, which is basically every page load. Adding a `useEffect` would allow us to update persistence while the user is still on the same page. Are there cases where we would want that?

Reply:
I would do it as a bit of a safety, in case for some reason we ever want to have the property be controlled.

Otherwise I would call out that it's not a reactive property

Reply:
Gotcha https://github.com/getsentry/sentry/pull/49048/commits/256ea4c751dc6bdf9205b0a8c78f8304da94679b